### DISCUSSION TITLE
High level documentation of Boundary Conditions

### DISCUSSION THREAD

**Discussion by ttruster** [idaholab/moose#23291] [Q&A General] [answered]
As I was working on my own documentation for a pull request, I noticed that all the other MOOSE syntax pages except for https://mooseframework.inl.gov/syntax/BCs/index.html have a good or even great discussion of the types of classes derived from that syntax. E.g. Variables, Kernels, Constraints, Interface Kernels, ...

Other than the example at https://mooseframework.inl.gov/getting_started/examples_and_tutorials/examples/ex04_bcs.html and some of the videos, I haven't found a nice high-level list of the types of BC in MOOSE and how to add new ones. If I'm missing the link, let me know so I can bookmark it.

I'm not volunteering since I'm rather rather new here, but I'll throw out some ideas. I think like half a page is good enough to categorize them; like what is at the https://mooseframework.inl.gov/syntax/ICs/index.html page. Mentioning the difference (input file and PDE-treatment) of Dirichlet and Neumann BC (and Robin), and nodal versus integrated BCs.

**Comment by GiudGiud** [comment, marked as answer]
Hello

You are very much right. We are missing syntax-documentation for some of the core systems of MOOSE
This would be the page for it 
https://mooseframework.inl.gov/syntax/BCs/index.html

We have spent a lot of time to document all the framework objects. The syntax / core systems can be a little tougher to document because of all the things one has to know about the system, hence the delay. I will try to find the time.

Guillaume
